### PR TITLE
PCESA-1801 Added support for sending CNM response to 1 or more SNS topics or Kinesis streams

### PR DESCRIPTION
**Pull request by ntquach**

### CODE REVIEW

File: src/main/java/gov/nasa/cumulus/SenderFactory.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+/**
+ * Factory class to return Sender based on type
+ */
+public class SenderFactory {
+    public static Sender getSender(String region, String type) {
+        if ("kinesis".equalsIgnoreCase(type)) {
+            return new KinesisSender(region);
+        } else if ("sns".equalsIgnoreCase(type)) {
+            return new SNSSender(region);
+        } else {
```
Comment (mgangl) [inline src/main/java/gov/nasa/cumulus/SenderFactory.java:12]:
```suggestion

/**
 * Factory class to return Sender based on type
 */
public class SenderFactory {
    public static final string METHOD_KINESIS = "kinesis"; 
    public static final string METHOD_SNS = "sns";   
    
    public static Sender getSender(String region, String type) {
        if (METHOD_KINESIS.equalsIgnoreCase(type)) {
            return new KinesisSender(region);
        } else if (METHOD_SNS.equalsIgnoreCase(type)) {
            return new SNSSender(region);
        } else {
```

Comment (mgangl) [reply src/main/java/gov/nasa/cumulus/SenderFactory.java:12]: might even be pulled outside senderFactory, but at least it's available outside if needed.

File: src/main/java/gov/nasa/cumulus/CNMResponse.java [hunk 2]
```diff
@@ -217,15 +150,14 @@ public String PerformFunction(String input, Context context) throws Exception {
...
-			CNMResponse.sendMessageKinesis(output, region, endpoint);
-		}else if(method != null && method.equals("sns")){
-			CNMResponse.sendMessageSNS(output, region, endpoint);
+		JsonElement responseEndpoint = inputConfig.get("response-endpoint");
+		if (method != null) {
+			if (responseEndpoint.isJsonArray()) {
+				SenderFactory.getSender(region, method).sendMessage(output, responseEndpoint.getAsJsonArray());
+			} else {
+				SenderFactory.getSender(region, method).sendMessage(output, responseEndpoint.getAsString());
+			}
```
Comment (mgangl) [inline src/main/java/gov/nasa/cumulus/CNMResponse.java:160]: This was clever. +1

**Comment by mgangl** [comment]
Testing coverage at 56%. Mostly ignoring kinesis. Have we ever sent to kinesis in a response? Might take out that code.

Also, should probably create a ticket to mock out the AWS portions instead of creating a topic and sending it. Did i write that originally?

**Comment by ntquach** [comment]
We'll need to survey other DAACs to see if anyone is sending response to kinesis before pulling out kinesis code. Cumulus CNM workflow example is using kinesis in CnmResponse 
 task https://nasa.github.io/cumulus/docs/data-cookbooks/cnm-workflow

New ticket created to capture mocking out AWS dependencies in unit tests: PCESA-2153

**Pull request merged**